File: languages/go/entity/key/access_control_totp.go
package key

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// AccessControlTotp represents a Time-based One-Time Password ([TOTP]) code used for access control.
//
// [TOTP]: https://datatracker.ietf.org/doc/html/rfc6238
type AccessControlTotp struct {
	Code string
}

// NewAccessControlTotp creates a new AccessControlTotp instance with the provided TOTP code.
func NewAccessControlTotp(code string) *AccessControlTotp {
	return &AccessControlTotp{
		Code: code,
	}
}

func (a AccessControlTotp) ToProto() *proto.AccessControlTotp {
	return &proto.AccessControlTotp{
		Code: a.Code,
	}
}

File: languages/js/src/entity/key/access_control_totp.ts
import * as proto from "../../bridge/proto/bloock_keys_entities";

/**
 * Represents a Time-based One-Time Password ([TOTP](https://datatracker.ietf.org/doc/html/rfc6238)) code used for access control.
 */
export class AccessControlTotp {
  code: string;

  /**
   * Constructs an AccessControlTotp object with the specified parameters.
   * @param code
   */
  constructor(code: string) {
    this.code = code;
  }

  public toProto(): proto.AccessControlTotp {
    return proto.AccessControlTotp.fromPartial({
      code: this.code
    });
  }
}
